src/Controller.js

```diff
@@ -0,0 +1,123 @@
+import FunctionTree from 'function-tree'
+import Module from './Module'
+import DefaultModel from './DefaultModel'
+import {ensurePath, isDeveloping, throwError} from './utils'
+import VerifyInputProvider from './providers/VerifyInput'
+import ContextProvider from 'function-tree/providers/Context'
+import StateProvider from './providers/State'
+import DebuggerProvider from './providers/Debugger'
+import ControllerProvider from './providers/Controller'
+import {EventEmitter} from 'events'
+import {dependencyStore as computedDependencyStore} from './Computed'
+
+/*
+  The controller is where everything is attached. The devtools
+  and router is attached directly. Also a top level module is created.
+  The controller creates the function tree that will run all signals,
+  based on top level providers and providers defined in modules
+*/
+class Controller extends EventEmitter {
+  constructor({state = {}, routes = {}, signals = {}, providers = [], modules = {}, router, devtools}) {
+    super()
+    this.flush = this.flush.bind(this)
+    this.devtools = devtools
+    this.model = new DefaultModel({})
+    this.module = new Module([], () => ({
+      state,
+      routes,
+      signals,
+      modules
+    }), {controller: this})
+    this.router = router ? router(this.module.getRoutes(), this) : null
+
+    const allProviders = (
+      this.router ? [
+        this.router.provider
+      ] : []
+    ).concat((
+      this.devtools ? [
+        DebuggerProvider(this.devtools)
+      ] : []
+    )).concat((
+      isDeveloping() ? [
+        VerifyInputProvider
+      ] : []
+    )).concat(
+      ControllerProvider(this),
+      StateProvider(this.model)
+    ).concat(
+      providers.concat(this.module.getProviders())
+        .map(provider => typeof provider === 'function' ? provider : ContextProvider(provider))
+    )
+
+    this.runTree = new FunctionTree(allProviders)
+    this.runTree.on('asyncFunction', this.flush)
+    this.runTree.on('end', this.flush)
+
+    if (this.devtools) {
```

Comment:
I had the same thoughts....providers via `function-tree` largely replace community-modules. And are  better in just about every way, IMO. More flexible, simpler api, less unique to cerebral...just including any library on context.